src/store/modules/milk/selectors.ts

```diff
@@ -0,0 +1,127 @@
+import { createSelector } from 'reselect';
+import { MilkEntry } from './types';
+import { State } from '../../types';
+
+import * as moment from 'moment';
+
+const getMilkEntries = (state: State) => state.milk.rows;
+const getCurrentFarmerUUID = (state: State) => state.currentFarmer.currentFarmerUUID;
+const radix: number = 10;
+
+/************Selectors for all milk entries (used on Home page) ***************/
+/**Selector to calculate the current days milk collection */
+export const getDaysMilkTotal = createSelector(
+  [getMilkEntries],
+  (milkEntries: MilkEntry[]) => milkEntries.reduce((sum: number, entry: MilkEntry) =>
+    inSameDay(entry.datetime) ? sum + parseInt(entry.volume, radix) : sum + 0, 0).toFixed(0));
```

Comment:
Maybe I'm not understanding correctly but I don't think this is the correct logic for this step. `reduce` takes two arguments, and both are of type `MilkEntry`. So any filtering and processing should be done before your reduce step. This snippet might help

```
milkEntries
  .filter(m => isSameDay(m.datetime))
  .map(m => parseInt(m.volume, radix))
  .reduce((a, b) => a + b)
  .toFixed(0)
```

Each of these steps (filter, map, reduce) should have it's own selector for it to keep things simple. So like a 
`getMilkEntriesFromToday`
`getVolumeEntriesFromToday`
`calculateVolumeSumToday`
with each building a new simple step on top of the previous one

Reply:
Reduce initializes to the first arg to milkentries unless you use an optional arg to initialize it to something else. So in this case I've initialized it to zero so I can use it to sum the applicable volumes

Reply:
Ohh I see, thanks I learned something! I think it would still be a good idea to break this down more into clear steps so it's easier to understand

Reply:
can do

Reply:
How do you feel about me simplifying it in the coming week and keeping it like this for the morning demo?

Reply:
Yeah sure that's fine, as long as it works for now